modules/distributor/artificial_delay.go

```diff
@@ -6,7 +6,11 @@ import (
 )
 
 func (d *Distributor) padWithArtificialDelay(reqStart time.Time, userID string) {
-	artificialDelay := d.overrides.IngestionArtificialDelay(userID)
+	artificialDelay := d.cfg.ArtificialDelay
+	if artificialDelayOverride, ok := d.overrides.IngestionArtificialDelay(userID); ok {
```

Comment:
Should not check that` d.overrides.IngestionArtificialDelay(userID)` is not 0?. If not it can overwrite the default value (not sure if that is the intended behaviour).

We do something similar here:

https://github.com/grafana/tempo/blob/ff83648dc7415e109c9649a977c9215f0e2f0bf3/modules/frontend/metrics_query_range_sharder.go#L326-L334

Reply:
Do we allow a user override with value 0 to override the default one?. If that's the case, then this is fine. In the test case that I propose below we find that scenario